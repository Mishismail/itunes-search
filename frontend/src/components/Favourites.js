//This code is a React functional component. It receives an array of favorite items (favorites) and a function (handleRemoveFromFavorites) as props. 
//The component renders a list of favorite items with a "Remove" button next to each item, allowing users to remove items from their favorites list.

import React from 'react';
import { ListGroup, Button } from 'react-bootstrap';

function Favourites({ favorites, handleRemoveFromFavorites }) {
  return (
    <div>
      <h2>Favourites</h2>
      <ListGroup>
        {favorites.map((favorite) => (
          <ListGroup.Item key={favorite.trackId} className="searched-item">
            {favorite.trackName}
            <Button
              variant="outline-danger"
              size="sm"
              className="ml-2"
              onClick={() => handleRemoveFromFavorites(favorite)}
            >
              Remove
            </Button>
          </ListGroup.Item>
        ))}
      </ListGroup>
    </div>
  );
}

export default Favourites;