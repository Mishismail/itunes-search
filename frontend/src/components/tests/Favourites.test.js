// src/components/__tests__/Favourites.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Favourites from '../Favourites';

it('renders correctly', () => {
  const favorites = [
    { trackId: 1, trackName: 'Test Track 1' },
    { trackId: 2, trackName: 'Test Track 2' }
  ];

  const tree = renderer.create(
    <Favourites favorites={favorites} handleRemoveFromFavorites={jest.fn()} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});