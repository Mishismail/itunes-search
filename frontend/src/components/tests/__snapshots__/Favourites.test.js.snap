// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Favourites
  </h2>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <div
      className="searched-item list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      Test Track 1
      <button
        className="ml-2 btn btn-outline-danger btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
    <div
      className="searched-item list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      Test Track 2
      <button
        className="ml-2 btn btn-outline-danger btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
